tests/resources/test_gemd_resource.py

```diff
@@ -120,6 +145,51 @@ def test_register(gemd_collection):
             assert pair[1] == registered.uids[pair[0]]
 
 
+def test_gemd_register(gemd_collection):
+    """Check that register routes to the correct collections"""
```

Comment:
This is also checking that register appends a uid to the returned object, but not the original object, yes?

Reply:
I believe the intended behavior in GEMD object registration is to edit the incoming objects such that they also contain new identifiers.  For objects with no UIDs, the act of serialization (all client-side) requires that they have at least one UID and CitrineIDs are assigned client side.  That alone would modify the objects.

This may be inconsistent with other behaviors in the library, and is certainly inconsistent with a `no side effects` design paradigm, but gemd-python doesn't follow a `no side effects` policy around serialization.